### ISSUE TITLE
Method Caller PID and disconnection from session dbus

### ISSUE THREAD

**Issue by yousefvand** [godbus/dbus#248]
Considering [server](https://github.com/godbus/dbus/blob/master/_examples/server.go) example and process `A` and `B` if one of them calls `Foo` method, how can I find it's `PID` and say it was `A` for example.
If both `A` and `B` are connected to session dbus and `B` gets disconnected from session dbus how can I inform about that (this is possible in C++ dbus library).

**Comment by guelfey** [comment]
> Considering server example and process A and B if one of them calls Foo method, how can I find it's PID and say it was A for example.

This is right now not possible with the "high-level" server mechanisms, but you can use [Eavesdrop](https://pkg.go.dev/github.com/godbus/dbus?utm_source=godoc#Conn.Eavesdrop) to get the sender of messages as well, and then query the process id of the sender using [org.freedesktop.DBus.GetConnectionUnixProcessID](https://dbus.freedesktop.org/doc/dbus-specification.html#bus-messages-get-connection-unix-process-id).

> If both A and B are connected to session dbus and B gets disconnected from session dbus how can I inform about that (this is possible in C++ dbus library).

Not sure if I understand you correctly; my best guess would be https://dbus.freedesktop.org/doc/dbus-specification.html#bus-messages-name-lost. If you have the C++ library ready, do you have a link to the documentation of this and/or some example code for what you mean?
